examples/bri-3/src/bri/communication/agents/messaging.agent.ts

```diff
@@ -26,8 +29,92 @@ export class MessagingAgent implements OnApplicationBootstrap {
 
   private onNewMessageReceived(rawMessage: string): void {
     this.logger.logInfo(
-      `MessagingListenerAgent: New raw message received: ${rawMessage}. Dispatching ProcessInboundMessageCommand`,
+      `MessagingListenerAgent: New raw message received: ${rawMessage}. Trying to parse into a Bpi Message`,
+    );
+
+    const [newBpiMessageCandidate, validationErrors] =
+      this.validateBpiMessageFormat(rawMessage);
+
+    if (validationErrors.length > 0) {
+      this.logger.logError(
+        `MessagingListenerAgent: Failed parsing message into Bpi Message. Errors: ${validationErrors.join(
+          '; ',
+        )}`,
+      );
+
+      return;
+    }
+
+    this.commandBus.execute(
+      new ProcessInboundBpiMessageCommand(
+        newBpiMessageCandidate.id,
+        newBpiMessageCandidate.from,
+        newBpiMessageCandidate.to,
+        JSON.stringify(newBpiMessageCandidate.content),
+        newBpiMessageCandidate.signature,
+        newBpiMessageCandidate.type,
+      ),
+    );
+  }
+
+  public validateBpiMessageFormat(
+    rawMessage: string,
+  ): [CreateBpiMessageDto, string[]] {
+    const errors: string[] = [];
+    let newBpiMessageCandidate: CreateBpiMessageDto;
+
+    try {
+      newBpiMessageCandidate = this.parseJsonOrThrow(rawMessage);
+    } catch (e) {
+      errors.push(`${rawMessage} is not valid JSON. Error: ${e}`);
+      return [newBpiMessageCandidate, errors];
+    }
+
+    if (!validate(newBpiMessageCandidate.id)) {
+      errors.push(`id: ${newBpiMessageCandidate.id} is not a valid UUID`);
+    }
+
+    if (!validate(newBpiMessageCandidate.from)) {
+      errors.push(`from: ${newBpiMessageCandidate.from} is not a valid UUID`);
+    }
+
+    if (!validate(newBpiMessageCandidate.to)) {
+      errors.push(`to: ${newBpiMessageCandidate.to} is not a valid UUID`);
+    }
+
+    if (
+      !newBpiMessageCandidate.content ||
+      typeof newBpiMessageCandidate.content !== 'object'
+    ) {
+      errors.push(
+        `content: ${newBpiMessageCandidate.content} is not valid JSON.`,
+      );
+    }
+
+    if (
+      !newBpiMessageCandidate.signature ||
+      newBpiMessageCandidate.signature.length === 0
+    ) {
+      errors.push('signature is empty');
+    }
+
+    if (newBpiMessageCandidate.type !== BpiMessageType.Info) {
+      errors.push(`type: ${newBpiMessageCandidate.type} is unknown`);
+    }
+
+    return [newBpiMessageCandidate, errors];
```

Comment:
same thing in other PR, this looks more like a golang style of programming, shouldnt we return some object here:
{
candidate,
errors
}
instead?

Reply:
We can go either with this approach or an object approach. I like this one more because of the nice way of concise way of describing the return types of the function: _[CreateBpiMessageDto, string[]]_.